include/AST/Stmt/ForStmt.hpp

```diff
@@ -0,0 +1,61 @@
+#ifndef GLU_AST_STMT_WHILESTMT_HPP
+#define GLU_AST_STMT_WHILESTMT_HPP
+
+#include "ASTNode.hpp"
+
+#include "ASTNode.hpp"
+#include "Stmt/CompoundStmt.hpp"
+
+namespace glu::ast {
+
+/// @class ForStmt
+/// @brief Represents a for statement in the AST.
+///
+/// This class inherits from StmtBase and encapsulates the details of a for
+/// statement.
+class ForStmt : public StmtBase {
+    /// @brief The init of the for statement.
+    ExprBase *_init;
```

Comment:
there is no init. the first node within a for statement is a kind of variable declaration. You can maybe create a node `ForBindingDecl` which inherits from `VarLetDecl` (which has a set name, inferred type, and _value = nullptr).